vowpalwabbit/parse_primitives.cc

```diff
@@ -53,6 +53,82 @@ std::vector<std::string> escaped_tokenize(char delim, VW::string_view s, bool al
   return tokens;
 }
 
+bool is_delim(char c) { return c == ' '; }
+
+bool is_quote(char c) { return c == '"' || c == '\''; }
+
+bool is_escape_char(char c) { return c == '\\'; }
+
+template <typename It>
+char unescape_char(It char_to_unescape_it, It end)
+{
+  if (char_to_unescape_it == end) { THROW("unescape_char: unexpected end of string while unescaping"); }
+
+  char c = *char_to_unescape_it;
+
+  if (c == 'n') { return '\n'; }
```

Comment:
Should there be more special cases here like `\t`

Reply:
Doesn't seem super necessary but I guess it doesnt hurt to add